numpy/core/src/umath/string_ufuncs.cpp

```diff
@@ -355,6 +401,76 @@ init_comparison(PyObject *umath)
 }
 
 
+static NPY_CASTING
+string_addition_resolve_descriptors(
+        PyArrayMethodObject *NPY_UNUSED(self),
+        PyArray_DTypeMeta *NPY_UNUSED(dtypes[3]),
+        PyArray_Descr *given_descrs[3],
+        PyArray_Descr *loop_descrs[3],
+        npy_intp *NPY_UNUSED(view_offset))
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = given_descrs[0];
+    Py_INCREF(given_descrs[1]);
```

Comment:
This isn't actually correct.  We need to ensure native byte-order for all dtypes.  (I suspect there is an existing pattern here, but not sure.)

Reply:
Changed this in https://github.com/numpy/numpy/pull/24858/commits/b49a6df5c5bff39caade0dfc56915597cd107136 to call `NPY_DT_CALL_ensure_canonical`, which I've seen being done elsewhere, but I'm not sure if that's the way.

Reply:
Yes, that works.